src/main/kotlin/systems/ajax/motrechko/airguardian/service/DroneService.kt

```diff
@@ -21,15 +25,20 @@ class DroneService(
 
     fun deleteDroneById(id: String): Mono<Unit> {
         return droneRepository
-            .findById(id)
-            .switchIfEmpty(Mono.error(DroneNotFoundException("Drone with id $id not found")))
-            .flatMap {
-                droneRepository.deleteById(id)
+            .deleteById(id)
+            .flatMap { deletedResult ->
+                if (deletedResult.deletedCount > 0) {
+                    Mono.just(Unit)
+                } else {
+                    Mono.error(DroneNotFoundException("Drone with id $id not found"))
+                }
             }
-            .then(Mono.empty())
     }
 
-    fun findDroneByStatus(droneStatus: DroneStatus): Flux<Drone> = droneRepository.findByStatus(droneStatus)
+    fun findDroneByStatus(droneStatus: DroneStatus): Mono<List<DroneResponse>> =
+        droneRepository.findAllByStatus(droneStatus)
+            .collectList()
+            .map { drones -> drones.map { it.toResponse() } }
```

Comment:
```suggestion
        droneRepository.findAllByStatus(droneStatus)
            .map { it.toResponse() }
            .collectList()
```
let's have mapping reactive